src/module/posts/posts.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable, NotFoundException, Post } from '@nestjs/common';
+import { ExceptionCodeName } from 'src/enum/exception-codes.enum';
+import { getRepository } from 'typeorm';
+import { RequestUserPayload } from '../auth/interface/request-user-payload.interface';
+import { UserService } from '../user/user.service';
+import { UnauthorizedException } from '@nestjs/common';
+import { Transactional } from 'typeorm-transactional-cls-hooked';
+import { CreatePostDto } from './dto/create-post.dto';
+import { UpdatePostDto } from './dto/update-post.dto';
+import { PostModel } from './posts.entity';
+
+
+@Injectable()
+export class PostsService {
+  constructor(
+    private userService: UserService
+  ){}
+
+  @Transactional()
+  async create(
+      createPostDto: CreatePostDto,
+      requestUserPayload: RequestUserPayload,
+  ): Promise<PostModel> {
+    const user = await this.userService.getOne({
+      where: {
+        id: requestUserPayload.id,
+      },
+    });
+    if (!user) {
+      throw new UnauthorizedException(ExceptionCodeName.INVALID_CREDENTIALS);
+    }
+    
+    const { title, content, date } = createPostDto;
+
+    const post = new PostModel();
+    post.title = title;
+    post.content = content;
+    post.user = user;
+    post.date = date;
+    return getRepository(PostModel).save(post);
+  }
+  
+  
+  @Transactional()
+  async update(
+    id: number, 
+    requestUserPayload: RequestUserPayload,
+    updatePostDto: UpdatePostDto,
+  ): Promise<PostModel> {
+    const user = await this.userService.getOne({
+      where: {
+        id: requestUserPayload.id,
+      },
+    });
+    if (!user) {
+      throw new UnauthorizedException(ExceptionCodeName.INVALID_CREDENTIALS);
+    }
+    const user_id = user.id;
+    const post = await this.getOneById(id, user_id);
+    const { title, content, date } = updatePostDto;
+    const newPost = {...post};
+    newPost.title = title || newPost.title;
+    newPost.content = content || newPost.content;
+    newPost.date = date || newPost.date;
+
+    return await getRepository(PostModel).save(newPost);
+  }
+  
+  
+  @Transactional()
+  async getAll(): Promise<PostModel[]> {
+    return getRepository(PostModel).createQueryBuilder('post_model')
+    .leftJoinAndSelect("post_model.user", "user")
+    .getMany();
+  }
+
+  
+  @Transactional()
+  async delete(id: number, requestUserPayload: RequestUserPayload): Promise<void> {
+    const user = await this.userService.getOne({
+      where: {
+        id: requestUserPayload.id,
+      },
+    });
+    if (!user) {
+      throw new UnauthorizedException(ExceptionCodeName.INVALID_CREDENTIALS);
+    }
+    const user_id = user.id;
+    const post = await this.getOneById(id, user_id);
+    getRepository(PostModel)
+      .createQueryBuilder()
+      .delete()
+      .where('id = :id', { id })
+      .andWhere('user_id = :user_id', {user_id})
+      .execute();
```

Comment:
No need for another query to database. Check getRepository(<Entity>).remove function